internal/pkg/service/buffer/sink/tablesink/storage/repository/file_repo.go

```diff
@@ -30,6 +30,26 @@ type FileRepository struct {
 	all     *Repository
 }
 
+// rotateSinkContext is an auxiliary struct to group arguments needed to rotate a Sink.
+type rotateSinkContext struct {
+	// Now timestamp, common for all parts of the atomic operation
+	Now time.Time
```

Comment:
I thought about how to improve the rotateAllIn method:
- How to shorten it.
- And how to get rid of those `continue` calls.

So I created a new `rotateSink` method, and since there would be many arguments, I wrapped them in the `rotateSinkContext` structure.